<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1044</width>
    <height>640</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Scene Graph</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="locale">
    <locale language="English" country="UnitedStates"/>
   </property>
   <layout class="QGridLayout" name="gridLayout" columnstretch="0,0,0,0,0,0,0">
    <item row="4" column="3">
     <widget class="QLabel" name="Scale">
      <property name="text">
       <string>Scale</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item row="2" column="5">
     <widget class="QLabel" name="Y">
      <property name="text">
       <string>Y</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="wordWrap">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item row="2" column="4">
     <widget class="QLabel" name="X">
      <property name="text">
       <string>X</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="4" column="5">
     <widget class="QDoubleSpinBox" name="scaleY">
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
     </widget>
    </item>
    <item row="3" column="4">
     <widget class="QDoubleSpinBox" name="transX">
      <property name="decimals">
       <number>4</number>
      </property>
      <property name="minimum">
       <double>-10.000000000000000</double>
      </property>
      <property name="maximum">
       <double>10.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
     </widget>
    </item>
    <item row="4" column="4">
     <widget class="QDoubleSpinBox" name="scaleX">
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
     </widget>
    </item>
    <item row="3" column="5">
     <widget class="QDoubleSpinBox" name="transY">
      <property name="decimals">
       <number>4</number>
      </property>
      <property name="minimum">
       <double>-99.000000000000000</double>
      </property>
      <property name="maximum">
       <double>99.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
     </widget>
    </item>
    <item row="5" column="4" colspan="2">
     <widget class="QDoubleSpinBox" name="rotAng">
      <property name="decimals">
       <number>2</number>
      </property>
      <property name="minimum">
       <double>-360.000000000000000</double>
      </property>
      <property name="maximum">
       <double>360.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>1.000000000000000</double>
      </property>
     </widget>
    </item>
    <item row="5" column="3">
     <widget class="QLabel" name="Rotation">
      <property name="text">
       <string>Rotation</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item row="3" column="3">
     <widget class="QLabel" name="Translation">
      <property name="text">
       <string>Translation</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item row="0" column="3" colspan="4">
     <widget class="QTreeWidget" name="treeWidget">
      <property name="rootIsDecorated">
       <bool>true</bool>
      </property>
      <property name="headerHidden">
       <bool>true</bool>
      </property>
      <column>
       <property name="text">
        <string notr="true">1</string>
       </property>
      </column>
     </widget>
    </item>
    <item row="8" column="3">
     <widget class="QLabel" name="OrientationAxis">
      <property name="text">
       <string>Orientation (Axis)</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item row="8" column="4">
     <widget class="QDoubleSpinBox" name="axisX">
      <property name="decimals">
       <number>4</number>
      </property>
      <property name="minimum">
       <double>-99.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
     </widget>
    </item>
    <item row="8" column="5">
     <widget class="QDoubleSpinBox" name="axisY">
      <property name="decimals">
       <number>4</number>
      </property>
      <property name="minimum">
       <double>-99.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
     </widget>
    </item>
    <item row="8" column="6">
     <widget class="QDoubleSpinBox" name="axisZ">
      <property name="decimals">
       <number>4</number>
      </property>
      <property name="minimum">
       <double>-99.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
     </widget>
    </item>
    <item row="10" column="4">
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeType">
       <enum>QSizePolicy::Preferred</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>10</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="6" column="3">
     <widget class="QLabel" name="Velocity">
      <property name="text">
       <string>Velocity</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item row="6" column="5">
     <widget class="QDoubleSpinBox" name="velocityY">
      <property name="decimals">
       <number>4</number>
      </property>
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
     </widget>
    </item>
    <item row="6" column="4">
     <widget class="QDoubleSpinBox" name="velocityX">
      <property name="decimals">
       <number>4</number>
      </property>
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
     </widget>
    </item>
    <item row="5" column="6">
     <widget class="QPushButton" name="rotNode">
      <property name="text">
       <string>RotationNode</string>
      </property>
     </widget>
    </item>
    <item row="3" column="6">
     <widget class="QPushButton" name="transNode">
      <property name="text">
       <string>TranslateNode</string>
      </property>
     </widget>
    </item>
    <item row="7" column="3">
     <widget class="QLabel" name="Acceleration">
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="text">
       <string>Acceleration</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item row="4" column="6">
     <widget class="QPushButton" name="scaleNode">
      <property name="text">
       <string>ScaleNode</string>
      </property>
     </widget>
    </item>
    <item row="1" column="4" colspan="2">
     <widget class="QPushButton" name="grow">
      <property name="text">
       <string>GROW</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0" rowspan="13" colspan="3">
     <widget class="MyGL" name="mygl">
      <property name="minimumSize">
       <size>
        <width>500</width>
        <height>0</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="7" column="4">
     <widget class="QDoubleSpinBox" name="accX">
      <property name="decimals">
       <number>4</number>
      </property>
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
     </widget>
    </item>
    <item row="7" column="5">
     <widget class="QDoubleSpinBox" name="accY">
      <property name="decimals">
       <number>4</number>
      </property>
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
     </widget>
    </item>
    <item row="9" column="4">
     <widget class="QDoubleSpinBox" name="angle">
      <property name="decimals">
       <number>4</number>
      </property>
      <property name="minimum">
       <double>-360.000000000000000</double>
      </property>
      <property name="maximum">
       <double>360.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
     </widget>
    </item>
    <item row="9" column="3">
     <widget class="QLabel" name="OrientationAngle">
      <property name="text">
       <string>(Angle)</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item row="9" column="5">
     <widget class="QLabel" name="Branching">
      <property name="text">
       <string> Branch Interval</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item row="9" column="6">
     <widget class="QSpinBox" name="branchFactor"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1044</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionQuit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyGL</class>
   <extends>QOpenGLWidget</extends>
   <header>mygl.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
